### DISCUSSION TITLE
Lesson6: AttributeError: 'NoneType' object has no attribute '_with_attr'

### DISCUSSION THREAD

**Discussion by jamsubzero** [smartcontractkit/full-blockchain-solidity-course-py#295] [Q&A] [answered]
Hi, I am getting this error while executing test_fund_me.py when running `brownie test` .
I first encountered this error while following the tutorial. 

Since I couldn't solve it, I decided to clone the lesson's repo to get the exact same source code, and to my surprise, I still got the same error.

Maybe there is something wrong with my setup, please help.

![image](https://user-images.githubusercontent.com/10081952/139688176-9ba18358-129b-450e-9520-688358a63f2d.png)

**Comment by cromewar** [comment, marked as answer]
Hello @jamsubzero I just cloned the repo and ran the test, which went fine as you can see bellow, this is estrange as is the exact same code but anyways I have some hints that could be handy.

1. Please execute the test like this: `brownie test -s -k test_only_owner_can_withdraw` and paste here the result.
2. `NoneType` means that instead of an instance of a Class, Object or whatever else you might got, you actually got `None`, for example like `fund_me` returns None or `bad_actor` is `None`. So I wonder if you cloned the entire repo and executed the test on it, or you cloned the repo and just replaced the test script into your code? If is the second one, you may have something else missing elsewhere.
3. May be your python is getting fussy, would you try to execute the project on a different python instance? you can manage virtual python versions using `virtualenv`, [here](https://help.dreamhost.com/hc/en-us/articles/115000695551-Installing-and-using-virtualenv-with-Python-3) you have a guide.

Attached Image (test running fine)

![image](https://user-images.githubusercontent.com/20199180/139701483-9e811457-480c-4603-9542-fa033bc59a96.png)

**Reply by jamsubzero** [reply]
@cromewar Thanks for the reply.

1. Here:
![image](https://user-images.githubusercontent.com/10081952/139762122-329a27fb-3d68-40de-9961-2260a3635600.png)

2. I have this issue both on my own code (since I'm following the tutorial) and the cloned repo. I don't usually clone the repo when I'm following the tutorial, I cloned it to test if there is something wrong with my code. That's also my first suspicion that there's a `None` somewhere, but I have tried running the test interactively (`--interactive`) and print those objects, and all of them have values. So there's really something wrong with my setup somewhere.
3.  Yeah, probably there is something wrong with my Python env or version incompatibility with brownie and python or whatnot. I will try your suggestion.

Thanks.

**Reply by cromewar** [reply]
Hi @jamsubzero, I hope we can find a solution to your problem.

1. I expected to get more information on that execution but it seems it just stops running, so nothing to do there.
2. Thanks for the clarification this helps us to discard problems related to the code itself, and also you used `--interactive` so we can be sure the code is not the problem.
3. This leads to this point, which I strongly thing is caused by your python installation, as you said you will try with other python instances (I would suggest use miniconda python).

Please don't forget to post your result here, this might help other members of the community if they face the same problem.

Best regards, Cromewar.

**Reply by jamsubzero** [reply]
A bit of update on this @cromewar 
I have tried using `virtualenv` and freshly install Brownie but I got the same error. I even use the same version of Python and Brownie as what you have in your screenshot above.

Additional details on the issue: 
Brownie only throws this error when calling a function that didn't pass the modifier's `require` condition (i.e. onlyOwner). Any other function call on the contract works fine. 

My hunch is that when a `revert` error is thrown, for some reason, Brownie cannot make sense of what is thrown by the EVM.

 Sadly, only I encountered this issue so there must be something wrong with my setup. For now, I will just continue with the rest of the tutorial.

**Reply by cromewar** [reply]
So it seems when the `require` condition is not fulfilled it throws an error but brownie don't show as it should, I highly recommend you to post this issue on official brownie repo https://github.com/eth-brownie/brownie on issues section.

I hope you will be able to continue the rest of the course without this problems

**Reply by jamsubzero** [reply]
Update on this. I finally got it to work.

I first examined the internals of brownie how it is throwing the null reference. It happens in `transaction.py#_raise_if_reverted line 421` and it seems it uses the exception thrown in `contract.py#__call__`. But for some reason that `exc` object is a `None` thus causing the issue on my end.

I have tried a lot of things like reinstalling brownie and using a venv for python since for me those are the obvious culprit/s.   
And lastly, I reinstalled ganache-cli but this time to a directory I owned (not by root). I'm not sure if this is the real issue since I am able to change a state variable in the contract.

To anyone who encounters the same, try the combination I did: 
* use virtual env, 
* install brownie to the venv, 
* reinstall your ganache-cli to the directory you own (or simply chown your node_modules dir)

**Reply by andreitoma8** [reply]
How can I reinstall ganache-cli to a dir I own? I ve got the same issue and there is no other option left for me.

**Reply by cromewar** [reply]
hello @andreitoma8 it depends on which OS you are using, for linux you can use EVM to install node and For MAC you can use Brew, this will allow you to install ganache on a non-root directory.

**Comment by gnoparus** [comment]
I've got the same problem. 

`AttributeError:` 'NoneType' object has no attribute '_with_attr'


After I did some research in this discussion forum, I tried reinstall ganache-cli as follows.

uninstall ganache-cli
```
npm uninstall ganache-cli
sudo npm uninstall ganache-cli --global

```
reinstall ganache-cli
```
sudo npm install ganache-cli --global
npm install ganache-cli

```
test again
`brownie test -k test_only_owner_can_withdraw`

And it worked.

`FAILED tests/test_fund_me.py::test_only_owner_can_withdraw - brownie.exceptions.VirtualMachineError: revert`

There are node_modules/ and package-lock.json appear on brownie_fund_me project path.

**Reply by cromewar** [reply]
Nice catch @gnoparus !!
